components/lidar.py

```diff
@@ -0,0 +1,61 @@
+import logging as log
+from wpilib import SerialPort
+
+class Lidar:
+    compatString = ["doof","testBoard"]
+    dist = 0
+
+    MXPserial: SerialPort
+
+    bufferArray = bytearray(18)
+
+    def on_enable(self):
+        self.read()
+
+    def checkAndRead(self, packet):
```

Comment:
Can you add a brief description on what checkAndRead() does and maybe drop in an example packet @Leaf-life can maybe help with this too.